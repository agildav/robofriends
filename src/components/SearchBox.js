import React from "react";
import "tachyons";

const SearchBox = ({ searchfield, searchChange }) => {
  return (
    <div className="pa1">
      <input
        className="bg-lightest-blue ba b--green pa3 mb4"
        type="search"
        placeholder="Search robots"
        onChange={searchChange}
      />
    </div>
  );
};

export default SearchBox;
