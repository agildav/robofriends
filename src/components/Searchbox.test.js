import { shallow } from "enzyme";
import React from "react";
import SearchBox from "./SearchBox";

test("SearchBox component is rendered", () => {
  expect.assertions(1);
  return expect(shallow(<SearchBox />)).toMatchSnapshot();
});
