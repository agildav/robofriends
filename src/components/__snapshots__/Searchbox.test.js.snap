// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component is rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="bg-lightest-blue ba b--green pa3 mb4"
        onChange={undefined}
        placeholder="Search robots"
        type="search"
/>,
      "className": "pa1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "bg-lightest-blue ba b--green pa3 mb4",
        "onChange": undefined,
        "placeholder": "Search robots",
        "type": "search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="bg-lightest-blue ba b--green pa3 mb4"
          onChange={undefined}
          placeholder="Search robots"
          type="search"
/>,
        "className": "pa1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bg-lightest-blue ba b--green pa3 mb4",
          "onChange": undefined,
          "placeholder": "Search robots",
          "type": "search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
